cogs/gpt_3_commands_and_converser.py

```diff
@@ -755,8 +776,8 @@ async def encapsulated_send(
                 self.generate_debug_message(prompt, response), self.debug_channel
             )
 
-            if user_id in self.awaiting_responses:
-                self.awaiting_responses.remove(user_id)
+            if ctx.author.id in self.awaiting_responses:

```

Comment:
Something with awaiting users stops the author from being able to converse in a case -- check the comment on the main thread

Reply:
It also seems to not allow the person who got locked out to have conversations in the future 
![image](https://user-images.githubusercontent.com/21161101/211163706-9bf76a56-69c4-4c5b-9fe4-46eaa3998b06.png)

Reply:
I've had an issue like this elsewhere too, if the API errors out like when it's overloaded it also doesn't remove you, requiring a bot restart

Reply:
> Do you think we should set the username or the server nickname as the name the bot uses stores in its history? Currently, it's their username.


I think we should go by server nickname so they have more flexibility

Reply:
> 

I'll fix this up

Reply:
Actually, I might've run into a different weird case, I tried to recreate this and it didn't happen again..

Reply:
I thing a sort of easy "sanity check" could be checking if the user is in awating_reaponse when end_conversation is ran and if they are remove them from it, what do you think?

Reply:
> I've had an issue like this elsewhere too, if the API errors out like when it's overloaded it also doesn't remove you, requiring a bot restart

but hmm, this should've been fixed earlier,  on error I have the following code:
![image](https://user-images.githubusercontent.com/21161101/211166077-b8c8b2c0-da16-4733-8b88-977ad3754c0a.png)

on an exception popping up, the user the exception happened for is attempted to be removed from awaiting responses, so it should purge unless there's some weird race condition, I'll look more into this

Reply:
It might be that I was on an older commit, here it is anyways
![image](https://user-images.githubusercontent.com/15952522/211166881-116ecc74-fa6b-4d48-bdca-f8089de9fdd3.png)

Reply:
Ah gotcha, lets make an issue if it happens again, because I think I put purges on both spots where errors pop up so that shouldn't happen anymore :)

Reply:
I found the isue, just got another one. ctx.user.id instead of ctx.author.id. discord.ApplicationContext has ctx.user and ctx.author which resolves to the same thing, a discord.Message only has ctx.author
![image](https://user-images.githubusercontent.com/15952522/211167444-514ca3f2-1093-4b43-abc0-5c742f2a4ee3.png)

Reply:
Ah, damn, that would do it!

Reply:
Fixing that up now, ill push up the fix to the main branch

Reply:
oops -- didn't see that you said you'll include a change to it here, sounds good!